client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/TransportFactory.java

```diff
@@ -1,95 +0,0 @@
-package org.infinispan.client.hotrod.impl.transport;
-
-import java.net.SocketAddress;
-import java.util.Collection;
-import java.util.Map;
-import java.util.Set;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.Consumer;
-
-import javax.net.ssl.SSLContext;
-
-import org.infinispan.client.hotrod.CacheTopologyInfo;
-import org.infinispan.client.hotrod.configuration.Configuration;
-import org.infinispan.client.hotrod.event.ClientListenerNotifier;
-import org.infinispan.client.hotrod.impl.consistenthash.ConsistentHash;
-import org.infinispan.client.hotrod.impl.consistenthash.ConsistentHashFactory;
-import org.infinispan.client.hotrod.impl.protocol.Codec;
-import org.infinispan.client.hotrod.impl.transport.tcp.TcpTransportFactory.ClusterSwitchStatus;
-import org.infinispan.commons.marshall.Marshaller;
-
-/**
- * Transport factory for building and managing {@link org.infinispan.client.hotrod.impl.transport.Transport} objects.
- *
- * @author Mircea.Markus@jboss.com
- * @since 4.1
- */
-public interface TransportFactory {
```

Comment:
TBH, I'm not sure what to think about this. On one hand this class resides in `impl` package but on the other it is a sort of contract and removing it might break backward compatibility. It is especially important because we expose a property which allows to swap implementations.

My personal vote would be to keep this class but I wonder what others think about it.

//cc @tristantarrant

Reply:
@slaskawi so you suggest keeping just the interface around, though not using it anywhere? I wouldn't have a problem with that. Though, I think that a completely custom implementation is extremely unlikely. There could be a tweak impl inheriting from TcpTransportFactory but we'd broke that anyway.

Btw., check out https://issues.jboss.org/browse/ISPN-8528.

Reply:
What I was trying to say is that the new implementation should implement this interface IMO.

Reply:
@slaskawi What's the point of keeping the interface when we would have to change the method signatures completely?

Reply:
As I mentioned before, it depends how do we treat this class. If we decided this is a private API, we can remove it in a minor release. However, if we decided that this is a public API, we can not remove it in a minor release.

Personally, I would treat it as a private API since it resides in `impl` package. But I'm really curious if @tristantarrant agrees with me here (and with you, since you're trying to remove this class).

Reply:
TcpTransportFactory is an actual API, since it is exposed in the configuration, although I doubt there are any practical situations where one would supply a custom implementation (unless, as @rvansa says, it is an extension of the existing one).
Therefore, for API preservation's sake, please keep the interface around, but log a deprecation warning